src/core/qgsuserprofilemanager.h

```diff
@@ -142,6 +154,57 @@ class CORE_EXPORT QgsUserProfileManager : public QObject
      */
     void setDefaultFromActive();
 
+    /**
+     * Returns the name of the lastly closed profile. Empty if its the first time QGIS has been run.
+     * \since QGIS 3.32
+     */
+    QString lastProfileName() const;
+
+    /**
+     * Updates the last closed profile name. Called when QGIS is closed.
+     * \since QGIS 3.32
+     */
+    void updateLastProfileName() SIP_SKIP;
+
+    /**
+    *    Returns the user profile selection policy.
+    *    \since QGIS 3.32
+    */
+    UserProfileSelectionPolicy userProfileSelectionPolicy() const;
+
+    /**
+      * Sets the user profile selection policy.
+      * \param policy The policy to use when selecting a user profile.
+      * \since QGIS 3.32
+    */
+    void setUserProfileSelectionPolicy( UserProfileSelectionPolicy policy );
+
+    /**
+     * Returns the icon size for the profile selector.
+     * \since QGIS 3.32
+     */
+    int profileSelectorIconSize() const;
```

Comment:
I don't think the icon size should be exposed in this core class -- it's a gui related setting. Could we just leave it as a qsettings value instead and not expose through api?

Reply:
Ok